client/src/components/export/dialogs/filamentExportDialog.tsx

```diff
@@ -0,0 +1,228 @@
+import { Col, Collapse, Form, Input, Modal, Switch } from "antd";
+import { IFilament } from "../../../pages/filaments/model";
+import { useSavedState } from "../../../utils/saveload";
+import { CSVExportOptions, exportAsCSV } from "../functions";
+import { useTranslate } from "@refinedev/core";
+
+interface FilamentExportDialog {
+  visible: boolean;
+  items: IFilament[];
+  onCancel: () => void;
+}
+
+const FilamentExportDialog: React.FC<FilamentExportDialog> = ({ visible, items, onCancel }) => {
+  const t = useTranslate();
+  const [csvDefaultCollapse, setCSVDefaultCollapse] = useSavedState<string[]>("export-collapseState-csv-filament", []);
+
+  const [csvExportOptions, setCSVExportOptions] = useSavedState<CSVExportOptions<IFilament>>("export-CSVOptions-filament", {
+    delimiter: ";",
+    includeHeaders: true,
+    filename: "Filament Export",
+  });
+
+
+  return (
+    <Modal
+      open={visible}
+      title={t("exporting.generic.title")}
+      onCancel={onCancel}
+      onOk={() => exportAsCSV(items, csvExportOptions)}
+      okText={t("exporting.generic.export")}
+    >
+      <Col>
+        <Col
+          style={{
+            whiteSpace: "pre-line",
+            marginBottom: "1em",
+          }}
+        >
+          {t("exporting.generic.description")}
+        </Col>
+        <Col>
+          {t("exporting.generic.csvOptions.title")}
+          <Form.Item label={t("exporting.generic.csvOptions.delimiter")}>
+            <Input
+              value={csvExportOptions.delimiter}
+              onChange={(e) => {
+                setCSVExportOptions({ ...csvExportOptions, delimiter: e.target.value });
+              }}
+            />
+          </Form.Item>
+          <Form.Item label={t("exporting.generic.csvOptions.includeHeaders")}>
+            <Switch
+              checked={csvExportOptions.includeHeaders}
+              onChange={(checked) => {
+                setCSVExportOptions({ ...csvExportOptions, includeHeaders: checked });
+              }}
+            />
+          </Form.Item>
+          <Collapse
+            defaultActiveKey={csvDefaultCollapse}
+            bordered={false}
+            ghost
+            onChange={(key) => {
+              if (Array.isArray(key)) {
+                setCSVDefaultCollapse(key);
+              }
+            }}
+          >
+            <Collapse.Panel header={t("exporting.generic.csvOptions.filamentFields")} key="1">
```

Comment:
I suggest an improvement to the selection UX, instead of using switches (or checkboxes), you can use toggle-able chip items, keep them inline which make this much more compact.

Here is an example from Monday
![image](https://github.com/Donkie/Spoolman/assets/101639089/887ce4f7-7760-42da-8bac-f26e5aa87cb1)

Reply:
Good idea, always forget these things exist.